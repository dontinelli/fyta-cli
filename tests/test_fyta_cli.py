"""Tests for fyta_cli."""

from aioresponses import aioresponses

from src.fyta_cli.fyta_client import FYTA_AUTH_URL
from src.fyta_cli.fyta_connector import FytaConnector
from src.fyta_cli.fyta_models import Credentials

from . import load_fixture


async def test_connection(
    responses: aioresponses,
) -> None:
    """Test connection."""
    responses.post(
        FYTA_AUTH_URL,
        status=200,
        body='{"message": "Unsupported Media Type"}',
    )

    fyta_connector = FytaConnector("example@example.com", "examplepassword")

    assert await fyta_connector.test_connection()

    assert fyta_connector.client.session is not None
    assert not fyta_connector.client.session.closed
    await fyta_connector.client.close()
    assert fyta_connector.client.session.closed


async def test_login(
    responses: aioresponses,
) -> None:
    """Test login."""
    responses.post(
        FYTA_AUTH_URL,
        status=200,
        body=load_fixture("login_response.json"),
    )

    fyta_connector = FytaConnector("example@example.com", "examplepassword")

    credentials: Credentials = await fyta_connector.login()

    assert credentials.access_token =="111111111111111111111111111111111111111"

    assert fyta_connector.client.session is not None
    assert not fyta_connector.client.session.closed
    await fyta_connector.client.close()
    assert fyta_connector.client.session.closed
