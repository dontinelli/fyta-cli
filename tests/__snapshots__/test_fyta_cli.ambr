# serializer version: 1
# name: test_get_plant_data[asyncio]
  dict({
    0: dict({
      'battery_level': 100.0,
      'last_updated': datetime.datetime(2023, 1, 1, 10, 10, tzinfo=datetime.timezone.utc),
      'light': 2.0,
      'light_status': <PlantMeasurementStatus.TOO_LOW: 1>,
      'low_battery': False,
      'moisture': 61.0,
      'moisture_status': <PlantMeasurementStatus.PERFECT: 3>,
      'name': 'Gummibaum',
      'online': True,
      'ph': None,
      'plant_id': 201,
      'plant_origin_path': '<url>',
      'plant_thumb_path': '<url>',
      'salinity': 1.0,
      'salinity_status': <PlantMeasurementStatus.LOW: 2>,
      'scientific_name': 'Ficus elastica',
      'sensor_available': True,
      'sensor_id': 'AA:AA:AA:2B:AF:F4',
      'sensor_update_available': False,
      'status': <PlantStatus.NEED_ATTENTION: 2>,
      'sw_version': '0.30.0',
      'temperature': 18.0,
      'temperature_status': <PlantMeasurementStatus.LOW: 2>,
    }),
    1: dict({
      'battery_level': 50.0,
      'last_updated': datetime.datetime(2023, 1, 1, 10, 10, tzinfo=datetime.timezone.utc),
      'light': None,
      'light_status': <PlantMeasurementStatus.NO_DATA: 0>,
      'low_battery': False,
      'moisture': 61.0,
      'moisture_status': <PlantMeasurementStatus.PERFECT: 3>,
      'name': 'Kakaobaum',
      'online': True,
      'ph': None,
      'plant_id': 201,
      'plant_origin_path': '<url>',
      'plant_thumb_path': '<url>',
      'salinity': 1.0,
      'salinity_status': <PlantMeasurementStatus.LOW: 2>,
      'scientific_name': 'Theobroma cacao',
      'sensor_available': True,
      'sensor_id': 'AA:AA:AA:2B:AF:F4',
      'sensor_update_available': False,
      'status': <PlantStatus.NEED_ATTENTION: 2>,
      'sw_version': '0.30.0',
      'temperature': None,
      'temperature_status': <PlantMeasurementStatus.NO_DATA: 0>,
    }),
  })
# ---
# name: test_get_plant_list[asyncio]
  dict({
  })
# ---
